datacube/api/core.py

```diff
@@ -1,21 +1,26 @@
 from __future__ import absolute_import, division, print_function
 
 import logging
-from itertools import groupby
+from itertools import groupby, repeat
 from collections import namedtuple, OrderedDict
 from math import ceil
 import warnings
 
+import os
+import uuid
 import pandas
 import numpy
 import xarray
+import SharedArray as sa
```

Comment:
I'd really like `SharedArray` to be an optional dependency. It's currently only available as a source package requiring compilation, which makes installing it much more difficult on Windows/Mac systems without a compiler.

It's also slowed down the review of all this work, since none of the existing environments available to me could run this branch.